ledger-tool/src/main.rs

```diff
@@ -2350,6 +2482,72 @@ fn main() {
                         system_monitor_service.join().unwrap();
                     }
                 }
+                ("simulate-block-production", Some(arg_matches)) => {
+                    let marker_file = simulate_block_production_marker_path(&ledger_path);
+                    if !marker_file.exists() {
+                        eprintln!(
+                            "simulate-block-production is for development purposes only. \
+                             It's thus a pretty destructive operation on \
+                             the ledger ({ledger_path:?}). \
+                             Create an empty file at {marker_file:?} if this is intentional."
+                        );
+                        exit(1);
+                    }
+
+                    let mut process_options = parse_process_options(&ledger_path, arg_matches);
+
+                    let banking_trace_events = load_banking_trace_events_or_exit(&ledger_path);
+                    process_options.hash_overrides =
+                        Some(banking_trace_events.hash_overrides().clone());
+
+                    let slot = value_t!(arg_matches, "first_simulated_slot", Slot).unwrap();
+                    let simulator = BankingSimulator::new(banking_trace_events, slot);
+                    let Some(parent_slot) = simulator.parent_slot() else {
+                        eprintln!(
+                            "Couldn't determine parent_slot of first_simulated_slot: {slot} \
+                             due to missing banking_trace_event data."
+                        );
+                        exit(1);
+                    };
+                    process_options.halt_at_slot = Some(parent_slot);
+
+                    let blockstore = Arc::new(open_blockstore(
+                        &ledger_path,
+                        arg_matches,
+                        AccessType::Primary, // needed for purging already existing simulated block shreds...
+                    ));
+                    let genesis_config = open_genesis_config_by(&ledger_path, arg_matches);
+                    let LoadAndProcessLedgerOutput { bank_forks, .. } =
+                        load_and_process_ledger_or_exit(
+                            arg_matches,
+                            &genesis_config,
+                            blockstore.clone(),
+                            process_options,
+                            None, // transaction status sender
+                        );
+
+                    let block_production_method = value_t!(
+                        arg_matches,
+                        "block_production_method",
+                        BlockProductionMethod
+                    )
+                    .unwrap_or_default();
```

Comment:
because of this https://github.com/solana-labs/solana/pull/30746#discussion_r1144901986, `--block-production-method` ended up not using clap's `.default_value()`, so `unwrap_or_default()` is intended here.

Reply:
Huh, I stand corrected. I bet we wouldn't this with newer CLAP but here we are haha; marking this resolved